### DISCUSSION TITLE
HTML Text editor

### DISCUSSION THREAD

**Discussion by fgilsc** [microsoft/ALAppExtensions#13004] [Q&A] [answered]
Hi,
I would like how it's made the HTML editor used on the Mail Body.

Regards

**Comment by JesperSchulz** [comment, marked as answer]
The HTML editor is an internal control, which currently is hardcoded to only work in the email editor. We are working on making that control available to all developers, as a general purpose control.

**Reply by fgilsc** [reply]
Thanks

**Reply by fgilsc** [reply]
How is this feature going?

**Reply by JesperSchulz** [reply]
We've run into issues, preventing us from releasing the editor as a standalone control. Solutions are being debated, but no concrete plans can get presented at this point in time.

**Reply by DanielGoehler** [reply]
@JesperSchulz This is very sad to hear. Would it be possible to create a new page "Html Editor" also hardcoded with that Html Editor Control with only procedures SetHtmlContent(Html: Text) and GetHtmlContent() : Text? 

You could sell this internally as there is a big demand in Shopify Connector to write formatted item descriptions.

**Reply by pri-kise** [reply]
@JesperSchulz  There also a discussion on twitter, about the lack of an HTML Editor for the Shopify Connetor.
https://publish.twitter.com/?query=https%3A%2F%2Ftwitter.com%2Fmizecaze%2Fstatus%2F1537319963794497537&widget=Tweet
If you add the HTML Editor on a generic CardPart, this could be used until you'll have some time to add it as real new control.
It would be perfect if we had the possibility to control which HTML Tags can be used in the editor. (e.g. to only support the HTML Tags for  RDLC Layouts)

**Reply by JesperSchulz** [reply]
I will meet with a few stakeholders after the summer break and we'll settle on a solution for the next major. It might be a hack, as described above, or it might be a more complete solution. But we've settled that we need a solution now - which is progress :-)

**Reply by pri-kise** [reply]
@JesperSchulz will we at least get a hack version for BC 22 (Release 2023 Wave 1)?

**Reply by JesperSchulz** [reply]
The full solution if off the table again - it again didn't make the cut.
The hack is still in my own backlog, even if I'm getting a little push back to release it. Hacks have a tendency to stick around and to replace the ambitions to do the right thing. If releasing the hack means I loose the chances of getting the full control released, I am hesitant.

**Reply by JesperSchulz** [reply]
My suggestion to implement a hack was declined. We DO want to deliver the real thing. Suggesting to implement the hack, sparked a new discussion, which resulted in me receiving confirmation that we're going to come up with a minimum viable solution for 2023 Wave 2 - and this time the work will be near the top of the backlog. I am confident, that we'll be able to deliver this time around.
We've got more and more scenarios, where a Rich Text control would come in handy, and the idea has hundreds of votes. I really think this time it will happen! 😊

**Reply by pri-kise** [reply]
@JesperSchulz it would be really helpful if the architecture of this addin will support some kind of limiting the available html options that are supported by the editor.
e.g. rdlc layouts or some other elements don't support all kind of html formatting or we don't want to enable all kind of formatting to the user on any place.

This doesn't have to be part of the first version, but it should be part of the arichtecture of this new Rich Text Control

**Comment by pri-kise** [comment]
It was announced at BCTechdays 2023 that we will get a RichTextEditor in a upcoming release. (No Details on the exact release date.)
<blockquote class="twitter-tweet"><p lang="en" dir="ltr">YES! We finally were able to deliver on my 3 year old “promise” of making the Rich Text editor from the email editor available as a standalone control 🥳 Sometimes, good things come to those who wait 😉 &mdash; Jesper Schulz-Wedde (MSFT) (@JesperSchulz) <a href="https://twitter.com/JesperSchulz/status/1672166960329637889?ref_src=twsrc%5Etfw">June 23, 2023</a></blockquote> 

I created a follow-up idea to limit the formatting that this editor could be used to specify text for RDLC layouts.

https://experience.dynamics.com/ideas/idea/?ideaid=659ed293-c514-ee11-913a-0003ff45ce96
